### PR TITLE
Reshape the submission object

### PR DESCRIPTION
**Pull request by diversemix**
Closes https://github.com/libero/reviewer/issues/650

### CODE REVIEW

File: src/domain/submission/services/submission-service.ts [hunk 1]
```diff
@@ -78,7 +77,7 @@ export class SubmissionService {
         if (!submission) {
             throw new Error('Unable to find submission with id: ' + id);
         }
-        submission.coverLetter = coverLetter;
+        submission.files.coverLetter = coverLetter;
```
Comment (erezmus) [inline src/domain/submission/services/submission-service.ts]: This doesn't look right

File: src/application/wizard/service.ts [hunk 2]
```diff
@@ -201,7 +201,10 @@ export class WizardService {
         const supportingFiles = (await this.fileService.getSupportingFiles(submissionId)).filter(
             file => !file.isCancelled() && !file.isDeleted(),
         );
-
-        return { ...submission, manuscriptFile, supportingFiles } as Submission;
+        const files: FileDetails = {
+            manuscriptFile,
+            supportingFiles,
+        };
+        return { ...submission, files } as Submission;
```
Comment (norisalsaadie) [inline src/application/wizard/service.ts]: Not related to this PR, and I only just noticed this, but we shouldn't be doing. We should be using the constructor

Comment (norisalsaadie) [reply src/application/wizard/service.ts]: Just tested this. This should raise a compiler error due to required properties, but it doesn't: 
`return { id: submission.id } as Submission;`

File: src/domain/submission/services/submission-service.ts (hunk 1 above)
Comment (diversemix) [reply src/domain/submission/services/submission-service.ts]: Not now - in the future it will be ;)

**Pull request merged**